tests/printer/expr/__snapshots__/render.spec.js.snap

```diff
@@ -2198,6 +2198,195 @@ let x =
+}
+
+// Mixed conditions, pattern start
+if let Some(x) = foo() {
+  doSomethingWithX(x)
+} else if n > 10 {
+  doSomethingWithForN()
+} else {
+  doSomethingElse()
+}
+
+// Mixed conditions, condition start
+if n > 10 {
+  doSomethingWithForN()
+} else if let Some(x) = foo() {
+  doSomethingWithX(x)
+} else {
+  doSomethingElse()
+}
+
+// Maintains attrs correctly
+if n > 10 {
+  @aa
+  doSomethingWithForN()
+} else if let @bb Some(@cc x) = @dd
+foo() {
+  @ee
+  doSomethingWithY(x)
+} else {
+  @ff
+  doSomethingElse()
+}
+
+if let Some(x) = foo() {
+  doSomethingWithX(x)
+} else if let Some(y) = bar() {
+  doSomethingWithY(y)
+} else if let Some(z) = baz() {
+  doSomethingWithZ(z)
+}
+
+// full destructuring
+if let Some(Thing(With({many: Internal([Components, q])}))) as p = foo() {
+  doSomethingWithE(e)
+}
+
+// Assignment
+let a = if let Some(x) = foo() {
+  x
+} else {
+  123
+}
+
+// Nesting
+let getZ = nested =>
+  if let Some(point) = nested.origin {
+    if let Some(z) = point.z {
+      z
+    } else {
+      0
+    }
+  } else {
+    0
+  }
+
+// Deep nesting
+let getZ = nested =>
+  if let Some(nested) = nested.a {
+    if let Some(nested) = nested.b {
+      if let Some(nested) = nested.c {
+        if let Some(nested) = nested.d {
+          if let Some(nested) = nested.e {
+            if let Some(nested) = nested.f {
+              if let Some(nested) = nested.g {
+                z
+              } else {
+                0
+              }
+            } else {
+              0
+            }
+          } else if let Some(nested) = nested.g {
+            z
+          } else {
+            0
+          }
+        } else if a {
+          b
+        } else {
+          c
+        }
+      } else {
+        0
+      }
+    } else {
+      0
+    }
+  } else {
+    0
+  }
+
+// Break testing
+if let Some(
+  suuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuperLongName,
+) = anotherSuuuuuuuuuuuuuuuuuuuuuuuuuuuperLongName() {
+  foo()
+}
+
+if let Some(
+  suuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuperLongName,
+) = anotherSuuuuuuuuuuuuuuuuuuuuuuuuuuuperLongNameanotherSuuuuuuuuuuuuuuuuuuuuuuuuuuuperLongName(
+  withSomeArgsThatAreAlsoLarge,
+  withManyArgsThatAreAlsoLarge,
+) {
+  foo()
+}
+
+if let {
+  suuuuuuuuuuuuuuuuuuuuuuuuuuuperLongName,
+  suuuuuuuuuuuuuuuuuuuuuuuuuuuperLongName2,
+  suuuuuuuuuuuuuuuuuuuuuuuuuuuperLongName3,
+} = buildMyRecord() {
+  foo()
+}
+
+// Complex nesting
+if let Some(Thing(With({many: Internal([Components, q])}))) as p = foo() {
+  if let Other(Thing(With({many: Internal([Components, q])}))) as p = foo() {
+    doSomethingElse(e)
+  }
+  doSomethingWithE(e)
+} else if let Some(Thing(With({
+  many: Internal([Components, q]),
+}))) as p = foo() {
+  doSomethingWithE(e)
+} else if let Some(Thing(With({
+  many: Internal([Components, q]),
+}))) as p = foo() {
+  doSomethingWithE(e)
+}
+
+// Ternary
+let a = b
+  ? if let Some(x) = foo() {
+      1
+    } else {
+      2
+    }
+  : 3
+
```

Comment:
This could alternately print as
```reason
let a = b ? if let Some(x) = foo() {
  1
} else {
  2
} : 3
```
But that would be a change to ternary printing, not `if let`

Reply:
Don't worry about this, it's not worth the time right now.